winml/lib/Telemetry/WinMLTelemetryHelper.cpp

```diff
@@ -178,4 +178,20 @@ void WinMLTelemetryHelper::SetNamedDimensionOverride(
       TraceLoggingWideString(name.c_str(), "dimensionName"),
       TraceLoggingInt32(value, "overrideValue"),
       TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES));
+}
+
+void WinMLTelemetryHelper::SetLearningModelDeviceKind(
+    int device_kind) {
+  if (!telemetry_enabled_)
+    return;
+  WinMLTraceLoggingWrite(
+      provider_,
+      "SetLearningModelDevice",
+      TraceLoggingKeyword(WINML_PROVIDER_KEYWORD_DEFAULT),
+      TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+      //Telemetry info
+      TraceLoggingUInt8(WINML_TLM_DEVICE_KIND_VERSION, "schemaVersion"),
+      // thread spinning info
+      TraceLoggingInt32(device_kind, "deviceKind"),
```

Comment:
Is int32 the right datatype we want to send up? Let's get sign off from Sabrina as well.

Reply:
Int32 is fine! It's consistent with the other telemetry values which are sent, and for FYI I am able to cast them to different types during the data processing.

Reply:
Sound good! I was just wondering if Int32 to represent deviceKind is better than a string like "Cpu" or "DirectX", "DirectXHighPerformance"

Reply:
that's a good point, since the telemetry is mostly accessed through reports and for DB performance reasons (int > strings), I believe Int32 is better. :)